src/main/java/org/jenkinsci/plugins/periodicbackup/S3.java

```diff
@@ -87,21 +91,27 @@ public S3(String bucket, boolean enabled, String tmpDir, String region, String c
     public Iterable<BackupObject> getAvailableBackups() {
         AmazonS3 client = AmazonUtil.getAmazonS3Client(region, credentialsId);
 
-        List<S3ObjectSummary> objectSummarys = client.listObjects(bucket).getObjectSummaries();
+        List<S3ObjectSummary> objectSummarys = client.listObjects(bucket, prefix).getObjectSummaries();
```

Comment:
does listObjects take null and empty, is it backwards compatible? why not something like StringUtils.isEmpty(prefix) ? client.listObjects(bucket) : client.listObjects(bucket, prefix)